connectivity/tests/encryption.go

```diff
@@ -59,8 +59,8 @@ func (s *podToPodEncryption) Run(ctx context.Context, t *check.Test) {
 		iface = "cilium_" + tunnelFeat.Mode // E.g. cilium_vxlan
 	} else {
 		cmd := []string{"/bin/sh", "-c",
-			fmt.Sprintf("ip -o r g %s from %s | grep -oP '(?<=dev )[^ ]+'",
-				server.Pod.Status.PodIP, client.Pod.Status.PodIP)}
```

Comment:
How did the tests pass?

Reply:
> In the direct routing mode, "ip r g $DST_POD_IP from $SRC_POD_IP"
  fails with "RTNETLINK answers: Network unreachable".

Do you know why? The revert means it may break on EKS.

Reply:
> How did the tests pass?

The direct routing was never tested. However, it's going to change in https://github.com/cilium/cilium/pull/22803.

> Do you know why? The revert means it may break on EKS.

My initial thought was that the rp_filter is to blame, but setting it to 0 didn't resolve the issue. For now my priority is to get many different DP configurations tested "on prem" (=Kind on LVH VMs). Once it's been done - managed K8s. However, it doesn't mean that others cannot pick it meanwhile.

Reply:
Got it. Might be worth creating an issue for that on this repository, just so we don't waste time when we start running this test on EKS in > 6months.

Reply:
> Might be worth creating an issue for that on this repository

Yep, I'm going to do in a bit.

Reply:
Just for posterity - https://github.com/cilium/cilium-cli/issues/1306.